app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue

```diff
@@ -109,25 +122,32 @@ export default {
     resetConfig() {
       this.viewer.reset();
     },
-    handleConfigUpdates() {
-      // NOTE: instead of compare the full object, we can destructure it and trigger the event only on those changing values we care
-      const config = this.getConfig()
-      if (JSON.stringify(this.prevConfig) !== JSON.stringify(config)) {
-        // don't emit event on the first load
-        if (this.prevConfig) {
-          this.$emit("showSaving")
-        }
-        this.prevConfig = config
-      }
-    },
     toggleConfigPerspective() {
-      this.$root.$emit('showSavedVizString', false)
-      this.viewer.toggleConfig()
+      this.showVizzEditor()
+      if (this.registrationDisabledAndUserIsLogged) {
+        this.$root.$emit('showSavedVizString', false)
+      }
     },
     hideConfigButton() {
       const configButtonPerspective = this.viewer.shadowRoot?.getElementById('config_button')
       configButtonPerspective.style.display = "none"
-    }
+    },
+    showVizzEditor() {
+      this.viewer.toggleConfig()
+      if (this.registrationDisabledAndUserIsLogged) {
+        const shadowRootPerspective = document.querySelector('perspective-viewer').shadowRoot
+        const selectVizPerspective = shadowRootPerspective.getElementById('vis_selector')
+        selectVizPerspective.addEventListener('click', () => {
+          this.$emit("showSaving")
+          this.$root.$emit('isVizModified', true)
+        })
+        const sidePanelPerspective = shadowRootPerspective.getElementById("side_panel");
+        sidePanelPerspective.addEventListener('click', () => {
+          this.$emit("showSaving")
+          this.$root.$emit('isVizModified', true)
+        })
+      }
```

Comment:
Reduce the number of listeners, adding the listener only to the parent, and check if the targets are those who trigger the event

Reply:
Done!